src/posix/tty.rs

```diff
@@ -55,7 +55,7 @@ impl TTYPort {
             Err(_) => return Err(super::error::from_raw_os_error(EINVAL))
         };
 
-        let fd = unsafe { libc::open(cstr.as_ptr(), O_RDWR | O_NOCTTY | O_NONBLOCK, 0) };
+        let fd = unsafe { libc::open(cstr.as_ptr() as *const c_char, O_RDWR | O_NOCTTY | O_NONBLOCK, 0) };
```

Comment:
That's odd. `CString::as_ptr()` is supposed to return a `*const c_char`: http://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr. The cast shouldn't be necessary.